File: Labs/Lab 4/Invertbst.py
class Invertbst:
    def invertTree(self, root):
        if root is None:
            return None
        left = self.invertTree(root.left)
        right = self.invertTree(root.right)
        root.left = right
        root.right = left
        return root
    class TreeNode(object):
        def __init__(self, val=0, left=None, right=None):
            self.val = val
            self.left = left
            self.right = right

# 1,2 8,3 4 5 6
invertbst = Invertbst()
root = invertbst.TreeNode(1)
root.left = invertbst.TreeNode(2)
root.right = invertbst.TreeNode(8)
root.left.left = invertbst.TreeNode(3)
root.left.right = invertbst.TreeNode(4)
root.right.left = invertbst.TreeNode(5)
root.right.right = invertbst.TreeNode(6)
invertedRoot = invertbst.invertTree(root)
print(invertedRoot.val)
print(invertedRoot.left.val, invertedRoot.right.val)
print(invertedRoot.left.left.val, invertedRoot.left.right.val, invertedRoot.right.left.val, invertedRoot.right.right.val)

File: Labs/Lab 4/Invertbst.java
public class Invertbst{
    public TreeNode invertTree(TreeNode root) {
        // check to see if node is null, return if so
        if (root == null) {
            return null;
        }
        // assign treenode vars to recursive calls to each side
        TreeNode left = invertTree(root.left);
        TreeNode right = invertTree(root.right);
        // swap sides
        root.left = right;
        root.right = left;
        return root;
    }
    public class TreeNode {

        int val;
   
        TreeNode left;
   
        TreeNode right;
   
        TreeNode() {}
   
        TreeNode(int val) { this.val = val; }
   
        TreeNode(int val, TreeNode left, TreeNode right) {
   
            this.val = val;
   
            this.left = left;
   
            this.right = right;
   
        }
    }
    public static void main(String[] args) {
        Invertbst invertbst = new Invertbst();
        // 1, 2, 8, 3, 4, 5, 6
        TreeNode root1 = invertbst.new TreeNode(1);
        root1.left = invertbst.new TreeNode(2);
        root1.right = invertbst.new TreeNode(8);
        root1.left.left = invertbst.new TreeNode(3);
        root1.left.right = invertbst.new TreeNode(4);
        root1.right.left = invertbst.new TreeNode(5);
        root1.right.right = invertbst.new TreeNode(6);
    
        TreeNode invertedRoot1 = invertbst.invertTree(root1);
        System.out.println(invertedRoot1.val);
        System.out.println(invertedRoot1.left.val+" "+invertedRoot1.right.val);
        System.out.println(invertedRoot1.left.left.val+" "+invertedRoot1.left.right.val+" "+invertedRoot1.right.left.val+" "+invertedRoot1.right.right.val);
    }
}